### PR TITLE
Direct3D interop sample

### PR DESCRIPTION
**Pull request by kekekeks**
direct3d

### CODE REVIEW

File: src/Avalonia.Controls/TopLevel.cs [hunk 1]
```diff
@@ -185,7 +185,7 @@ IStyleHost IStyleHost.StylingParent
         }
 
         /// <inheritdoc/>
-        IRenderTarget IRenderRoot.CreateRenderTarget()
```
Comment (grokys) [inline src/Avalonia.Controls/TopLevel.cs]: Does this need to be `public`? `protected` would make more sense IMO.

File: src/Avalonia.Controls/TopLevel.cs [hunk 2]
```diff
@@ -185,7 +185,7 @@ IStyleHost IStyleHost.StylingParent
...
+        public virtual IRenderTarget CreateRenderTarget()
```
Comment (grokys) [inline src/Avalonia.Controls/TopLevel.cs]: Does this need to be public? protected would make more sense IMO.

**Comment by jkoritzinsky** [comment]
Looks good to me! One more thing you could do is make a `SurfaceRenderTarget` class in SharpDX.Direct2D so people could use that instead of having to make their own RenderTarget implementation.

**Comment by kekekeks** [comment]
The surface is being recreated every time window is resized while IRenderTarget is being reused by renderer. I'm not sure how to properly workaround that. I could just throw RenderTargetCorrupted exception if it was disposed, but it doesn't look like a proper solution

File: src/Avalonia.Controls/TopLevel.cs (hunk 2 above)
Comment (kekekeks) [inline src/Avalonia.Controls/TopLevel.cs]: It was my initial thought, but complier was against that. Can't implement interface if member isn't public, blablabla

**Comment by jkoritzinsky** [comment]
Very good point. totally forgot about that. I think it's good the way it is then. I was thinking of a slightly different interop use case where that would work.

**Review by jkoritzinsky — APPROVED** [review]

File: src/Avalonia.Controls/TopLevel.cs (hunk 2 above)
Comment (grokys) [inline src/Avalonia.Controls/TopLevel.cs]: Could we add an explicit interface implementation that calls the protected method?

**Review by grokys — APPROVED** [review]

**Pull request merged**